sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1373,16 +1373,18 @@ object RepartitionByExpression {
  */
 case class RebalancePartitions(
     partitionExpressions: Seq[Expression],
-    child: LogicalPlan) extends UnaryNode {
-  override def maxRows: Option[Long] = child.maxRows
-  override def output: Seq[Attribute] = child.output
+    child: LogicalPlan) extends RepartitionOperation {
```

Comment:
This looks like a nice refactoring in order to make consistent and simplify.

Reply:
sorry for the late, I prefer not to do this. Since `RepartitionOperation` can affect the behavior of AQE, the semantics of `RepartitionOperation` have a little different from previous.

Currently, the `RepartitionOperation` behavior with AQE framwork follows:
* `Repartition` does not use AQE 
* `RepartitionByExpression` use AQE depends on `optNumPartitions`
*  `RebalancePartitions`(if extend) always use AQE

And the rule `CollapseRepartition` may not effective in all use case.